packages/nimble-components/src/table-column/icon/index.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable max-classes-per-file */
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { TableColumnEnumBase } from '../enum-base';
+import { styles } from '../enum-base/styles';
+import { template } from '../enum-base/template';
+import type { TableColumnValidity } from '../base/types';
+import { mixinGroupableColumnAPI } from '../mixins/groupable-column';
+import { mixinFixedWidthColumnAPI } from '../mixins/fixed-width-column';
+import { Mapping } from '../../mapping/base';
+import { MappingSpinner } from '../../mapping/spinner';
+import { MappingIcon } from '../../mapping/icon';
+import {
+    iconColumnValidityFlagNames,
+    TableColumnIconValidator
+} from './models/column-validator';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-icon': TableColumnIcon;
+    }
+}
+
+/**
+ * Table column that maps values to icons/spinners
+ */
+export class TableColumnIcon extends mixinGroupableColumnAPI(
+    mixinFixedWidthColumnAPI(TableColumnEnumBase)
+) {
+    protected get supportedMappingTypes(): readonly (typeof Mapping)[] {
+        return [
+            MappingIcon,
+            MappingSpinner
+        ] as const;
+    }
+
+    private readonly validator: TableColumnIconValidator = new TableColumnIconValidator(
+        this.columnInternals,
+        iconColumnValidityFlagNames
+    );
+
+    public override get validity(): TableColumnValidity {
+        return this.validator.getValidity();
+    }
+
+    /**
+     * @internal
+     */
+    public override handleChange(source: unknown, args: unknown): void {
+        super.handleChange(source, args);
+        if (source instanceof Mapping && typeof args === 'string') {
+            if (args === 'key') {
+                const keys = this.mappings?.map(x => x.key) ?? [];
+                this.validator.validateKeyValuesForType(keys, this.keyType);
+                const typedKeys = this.columnInternals.columnConfig?.convertedKeyMappings.map(
+                    x => x.key
+                ) ?? [];
+                this.validator.validateUniqueKeys(typedKeys);
+                this.validator.validateNoMissingKeys(this.mappings ?? []);
+            } else if (args === 'icon') {
+                this.validator.validateIconNames(this.mappings ?? []);
+            }
+        }
+    }
+
+    protected override mappingsChanged(): void {
+        super.mappingsChanged();
+        const keys = this.mappings?.map(x => x.key) ?? [];
+        this.validator.validateKeyValuesForType(keys, this.keyType);
```

Comment:
This logic looks very similar to the validation logic in `handleChange`. Can it be factored out into a shared function?

Reply:
Done